docs/architecture/adr/0022-server-authorization.md

```diff
@@ -0,0 +1,162 @@
+---
+adr: "0022"
+status: Proposed
+date: 2024-07-22
+tags: [server]
+---
+
+# 0022 - Authorization
+
+<AdrTable frontMatter={frontMatter}></AdrTable>
+
+## Context and Problem Statement
+
+Authorization logic decides whether a user is permitted to carry out an action. Our current
+authorization logic is dispersed throughout different layers of our server codebase and follows
+different patterns, some of which are no longer suitable for our changing permissions structures.
+
+We should decide on a single, consistent solution to how we manage authorization in our
+applications. This is primarily concerned with the main Bitwarden server and client code (including
+Password Manager, Admin Console and Provider Portal) but could be extended to other products.
+
+### Terminology
+
+- **Authentication** is the process of verifying a user's identity. ("Who are you?") It tells you
+  who they are, but not what they can do
+- **Authorization** is the process of determining who can access or modify a resource. ("Are you
+  allowed to do this?") It may or may not require authentication
+- **Validation** is the process of determining whether a request is valid according to business
+  logic. ("Can this be done?") Unlike authorization, it usually does not depend on the user's
+  identity (authentication) or permissions (authorization). For example, a vault item cannot be
+  restored unless it has first been (soft) deleted
+- **Resources** are data on the server that a user may try to access or modify - such as a vault
+  item, organization, collection, or group
+- **User verification** is a Bitwarden-specific UI flow aimed at re-establishing a user's identity
+  before they perform a sensitive action. e.g. entering your password again before viewing an api
+  key. This is an aspect of authentication, because it relates to verifying the user's identity
+
+### Current patterns
+
+To date, we have broadly used the following authorization patterns:
+
+- in the individual user context, we match the user ID in the JWT to the resource being accessed.
```

Comment:
```suggestion
- In the individual user context, we match the user ID in the JWT to the resource being accessed.
```

⛏️ since it's multiple sentences.